packages/template-retail-react-app/app/components/_app-config/index.jsx

```diff
@@ -37,18 +43,39 @@ const AppConfig = ({children, locals = {}}) => {
     const [basket, setBasket] = useState(null)
     const [customer, setCustomer] = useState(null)
 
+    const {correlationId} = useCorrelationId()
+    const headers = {
+        'correlation-id': correlationId
+    }
```

Comment:
Aside: one thing about adding correlation id in this manner, what's the impact of doing so? 🤔 

For example, because the correlation is updated on every page navigation, the AppConfig is now being re-rendered too. Previously, the AppConfig was rendered only once for the entire lifetime of the app.

Reply:
Thats a thinker .... The expensive operation of writing to the dom should not happen as the html isn't changing to my knowledge. Do we know the performance impact of that re-render?

Reply:
This is an expected behavior since the CID Provider wraps AppConfig. However, it should not cause a big performance impact on the app since it won't cause any re-rendering of its children unless there is a data change in different providers.